shardingsphere-infra/shardingsphere-infra-common/src/main/java/org/apache/shardingsphere/infra/security/AlgorithmSecureFactory.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.shardingsphere.infra.security;
+
+import org.apache.shardingsphere.infra.config.properties.ConfigurationPropertyKey;
+import org.apache.shardingsphere.infra.spi.ShardingSphereServiceLoader;
+
+import java.util.Collection;
+import java.util.Map;
+import java.util.Properties;
+import java.util.concurrent.ConcurrentHashMap;
+
+/**
+ * ShardingSphere algorithm secure factory.
+ */
+public final class AlgorithmSecureFactory {
+    private static AlgorithmSecureFactory instance = new AlgorithmSecureFactory();
+
+    private Map<String, AlgorithmSecure> algorithms = new ConcurrentHashMap<>(2);
+
+    private Map<String, String> secure2AlgType = new ConcurrentHashMap<>(2);
+
+    private boolean isInited;
```

Comment:
The class is used by multiple threads, is `isInited` thread safe?

Reply:
> The class is used by multiple threads, is `isInited` thread safe?

only init interface need single thread, other interface support mutil threads.so I have alread remove this field.